cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringClient.java

```diff
@@ -56,11 +56,11 @@ public final class TetheringClient {
     .create();
 
   private final RemoteAuthenticator remoteAuthenticator;
-  private final String instanceName;
+  private final CConfiguration cConf;
 
-  public TetheringClient(RemoteAuthenticator remoteAuthenticator, String instanceName) {
+  public TetheringClient(RemoteAuthenticator remoteAuthenticator, CConfiguration cConf) {
```

Comment:
It's still better to have a constructor that doesn't take a cconf, but takes the instanceName, connect timeout, and read timeout. That way we avoid looking up the same values multiple time from the conf, and we don't need to create a cconf in tests,etc. when we want to create a client. We can have another constructor that looks up the corresponding values from the cconf and calls the other constructor.

Reply:
fixed